sdk/src/types/block/address/bech.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::hash::Hash;
+
+use serde::{Deserialize, Serialize};
+
+use crate::types::block::address::Address;
+
+/// An address and its network type.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct Bech32 {
+    pub(crate) inner: Address,
+    #[serde(rename = "bech32Hrp")]
+    pub(crate) bech32_hrp: String,
+}
+
+impl AsRef<Address> for Bech32 {
+    fn as_ref(&self) -> &Address {
+        &self.inner
+    }
+}
+
+impl Bech32 {
+    /// Create a new address wrapper.
+    pub fn new(address: Address, bech32_hrp: String) -> Self {
+        Self {
+            inner: address,
+            bech32_hrp,
```

Comment:
Do you know if there is a function to validate HRPs in the bech32 crate? Otherwise I'd make new faillible and we should validate it

Reply:
couldnt find any

Reply:
About this, how do you want to validate? just 3 letters in front? How do you want to add a check? Just starting with 3 letters ?

Reply:
HRP is not necessarily only 3 letters

Reply:
> The human-readable part, which is intended to convey the type of data, or anything else that is relevant to the reader. This part MUST contain 1 to 83 US-ASCII characters, with each character having a value in the range [33-126]. HRP validity may be further restricted by specific applications.